state-chain/runtime/src/lib.rs

```diff
@@ -2243,6 +2243,17 @@ impl_runtime_apis! {
 				.try_into()
 				.map_err(|_| pallet_cf_swapping::Error::<Runtime>::BoostFeeTooHigh)?;
 
+			// Validate broker fee
+			if broker_commission < pallet_cf_swapping::Pallet::<Runtime>::get_minimum_fee_for_broker(&broker_id) {
+				return Err(DispatchErrorWithMessage::from("Broker commission is too low"));
+			}
+			let beneficiaries = pallet_cf_swapping::Pallet::<Runtime>::assemble_broker_fees(
+				broker_id.clone(),
+				broker_commission,
+				affiliate_fees.clone(),
+			);
+			pallet_cf_swapping::Pallet::<Runtime>::validate_broker_fees(&beneficiaries)?;
```

Comment:
In both cases where we use `assemble_broker_fees`, we immediately `validate_broker_fees` afterwards. We can just make `assemble_broker_fees` return a Result and always validate inside that fn?